src/index.js

```diff
@@ -54,6 +54,19 @@ function map(array, callback) {
   return result;
 };
 
+function reduce(array, callback, initial) {
+  let accumulator = initial == undefined ? array[0] : initial;
+  const startIndex = initial == undefined ? 1 : 0;
+  for (let i = startIndex; i < array.length; ++i) {
+    accumulator = callback(accumulator, array[i], i, array);
+  }
+  return accumulator;
+};
+
+function reduceRight(array, callback, initial) {
+  return reduce(array.reverse(), callback, initial);
```

Comment:
`reverse` changes the structure of the array it's called on, so you'll need to copy the array here.